examples/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -82,23 +84,39 @@ class TimeMonitor
   }
   void summarize(MPI_Comm comm, std::ostream &os = std::cout)
   {
-    // FIXME Haven't tried very hard to format the output.
     int comm_size;
     MPI_Comm_size(comm, &comm_size);
     int comm_rank;
     MPI_Comm_rank(comm, &comm_rank);
     int n_timers = _data.size();
+
+    os << std::left << std::scientific;
+
+    // Initialize with length of "Timer Name"
+    std::size_t const max_section_length = std::accumulate(
+        _data.begin(), _data.end(), std::size_t(10),
+        [](std::size_t current_max, entry_reference_type section) {
+          return std::max(current_max, section.first.size());
+        });
+
     if (comm_size == 1)
     {
-      os << "========================================\n\n";
+      std::string const header_without_timer_name = " | GlobalTime";
+      int const header_width =
+          max_section_length + std::max<int>(header_without_timer_name.size(),
+                                             std::cout.precision() + 9);
```

Comment:
What's 9?

Reply:
The width we need to print a floating point number in `scientific` format apart from the precision assuming the exponent needs two characters.